src/main/java/io/bf2/kafka/authorizer/CustomAclAuthorizer.java

```diff
@@ -246,6 +247,35 @@ void addAllowedListeners(Map<String, ?> configs) {
         }
     }
 
+    /**
+     * To avoid misleading users we delete principals that are explicitly targeted by CustomAclAuthorizer rules.
+     * It is possible for a user with ACL rules in Zookeeper to become a user managed by CustomAclAuthorizer, in which
+     * case we would never use or update rules for them in ZK. The old rulesremain, dangling, in ZK and visible through
+     * kafka ACL tooling. This would mislead users becuase there will be a mismatch between what the ZK rules say they
+     * should be permitted to do and the truth.
+     */
```

Comment:
```suggestion
    /**
     * To avoid misleading users we delete principals that are explicitly targeted by CustomAclAuthorizer rules.
     * It is possible for a user with ACL rules in the controller to become a user managed by CustomAclAuthorizer, in which
     * case we would never use or update rules for them in the controller. The old rulesremain, dangling, in the controller and visible through
     * kafka ACL tooling. This would mislead users becuase there will be a mismatch between what the controller rules say they
     * should be permitted to do and the truth.
     */
```

I don't think the problem is Zookeeper specific and will still be an issue once things switch to KRaft so use the generic controller term instead of Zookeeper/ZK.

Reply:
Changed to use `Kafka Control Plane` instead of Zookeeper/ZK